internal/provider/resource_vault_cluster.go

```diff
@@ -82,9 +82,17 @@ func resourceVaultCluster() *schema.Resource {
 				Type:             schema.TypeString,
 				Optional:         true,
 				ValidateDiagFunc: validateSemVer,
-				ForceNew:         true,
+				// Given that Plus tier clusters will have Primary and Secondary on same version, changing the min vault version on primary shouldn't be possible. We need to handle this during apply
+				// constraints and fail. Only Plus tier clusters with no secondary should be allowed to re-create a new cluster.
+				ForceNew: true,
```

Comment:
Could add a DiffSuppressFunc here ([example](https://github.com/hashicorp/terraform-provider-hcp/blob/main/internal/provider/resource_consul_cluster.go#L97)) to save the user from an unnecessary cluster recreation. It really only needs to recreate if the `min_version` is set to a version higher than the current version.